### PR TITLE
Test tools as a FluentAssertion extension

### PR DESCRIPTION
**Pull request by Corniel**
The (only) relevant assert in the test tools asserts on the uncommitted events that an event stream should have.

This is now (only) accessible as `buffer.Should().HaveUncommittedEvents()`.

### CODE REVIEW

File: test/Qowaiv.DomainModel.UnitTests/Models/Events/Complex.cs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+﻿namespace Models.Events;
+
+public sealed record class Complex(int Value, string Message, DateTime Date);
```
Comment (LauraXiulan) [inline test/Qowaiv.DomainModel.UnitTests/Models/Events/Complex.cs:3]: ComplexEvent? To keep it consistent with Other and Simple

File: test/Qowaiv.DomainModel.UnitTests/TestTools/Aggregate_should_have_uncommitted_events_specs.cs [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        .WithMessage(@"The uncommitted events where different than expected.
+[0] EmptyEvent
+[1] Expected: { Value: 23, Date: 06/30/1980 00:00:00 }
+    Actual:   { Value: 17, Date: 06/11/2017 00:00:00 }
+[2] SimpleEvent
+");
+    }
+
+    [Test]
+    public void extra_messages()
```
Comment (LauraXiulan) [inline test/Qowaiv.DomainModel.UnitTests/TestTools/Aggregate_should_have_uncommitted_events_specs.cs:101]: Extra_events

File: src/Qowaiv.DomainModel.TestTools/FluentAssertions/EventBufferAssertions.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+/// The type of the identifier of the event buffer.
+/// </typeparam>
+[Inheritable]
+[CLSCompliant(false)]
+public class EventBufferAssertions<TId> : GenericCollectionAssertions<EventBuffer<TId>, object, EventBufferAssertions<TId>>
+{
+    /// <summary>Initializes a new instance of the <see cref="EventBufferAssertions{TId}"/> class.</summary>
+    public EventBufferAssertions(EventBuffer<TId> subject) : base(subject) { }
+
+    /// <summary>Verifies that the <see cref="AggregateRoot{TAggregate}"/> has the expected uncommitted events.</summary>
```
Comment (LauraXiulan) [inline src/Qowaiv.DomainModel.TestTools/FluentAssertions/EventBufferAssertions.cs:19]: Aggregate?

Comment (Corniel) [reply src/Qowaiv.DomainModel.TestTools/FluentAssertions/EventBufferAssertions.cs:19]: What do you mean?

**Review by LauraXiulan — APPROVED** [review]

**Pull request merged**